### ISSUE TITLE
Beef's installation script triggers an error when executed in a Fedora 35 system

### ISSUE THREAD

**Issue by q2dg** [beefproject/beef#2352]
I've run 

```
sudo dnf install ruby ruby-devel
git clone https://github.com/beefproject/beef 
cd beef &&  sudo ./install
```
and I get this error

![Captura de 2022-03-13 12-34-25](https://user-images.githubusercontent.com/7274874/158057566-9d7bbdf1-cea8-4d9b-8d4f-d2d9479fe9fc.png)

Thanks a lot!

**Comment by theverygoodwebsite** [comment]
well then what is the problem?

**Comment by theverygoodwebsite** [comment]
so they error is that there is no gen file right?

**Comment by theverygoodwebsite** [comment]
*gem

**Comment by bcoles** [comment]
tl/dr:

```
sudo yum install ruby-devel
gem install ffi
bundle install
gem pristine --all
```

---

Confirmed. Tested install on Fedora 35 and hit the same issue.

![image](https://user-images.githubusercontent.com/434827/158078945-256260f9-146f-47f5-923e-acd0bc1821df.png)

This seems like a Fedora package issue - outside control of BeEF project. If system packages fail for some reason I'm not sure there's much we can do about this to fix the installer.

However, despite the error, simply running `bundle install` correctly installed dependencies, until `nio4r` :

![image](https://user-images.githubusercontent.com/434827/158079049-0ceebdbb-2487-4b0a-ab96-1d75c70b1811.png)

This is because nio4r requires ruby development packages:

```
sudo yum install ruby-devel
```

Running `bundle install` again progressed further, failing on installing `http-parser`, resolved by:

```
gem install ffi
```

Running `bundle install` completed successfully, but gems failed to load at BeEF startup. Resolved by:

```
sudo gem pristine --all
```

**Comment by q2dg** [comment]
Thanks a lot for your explanation and tutorial.
However, I'm stuck in the same error after doing what you propose.
Anyway, as it seems a Fedora problem I'll close the issue, waiting for an upstream solution
Thanks again
